packages/legend-application-studio/src/stores/editor/NewElementState.ts

```diff
@@ -471,23 +476,48 @@ export class NewPackageableConnectionDriver extends NewElementDriver<Packageable
 
 export class NewServiceDriver extends NewElementDriver<Service> {
   mappingOption?: PackageableElementOption<Mapping> | undefined;
+  runtimeOption?: RuntimeOption | undefined;
   constructor(editorStore: EditorStore) {
     super(editorStore);
 
     makeObservable(this, {
       mappingOption: observable,
+      runtimeOption: observable,
       setMappingOption: action,
+      setRuntimeOption: action,
       isValid: computed,
       createElement: action,
     });
     this.mappingOption =
       editorStore.graphManagerState.usableMappings.map(buildElementOption)[0];
+    this.runtimeOption =
+      editorStore.graphManagerState.usableMappings.map(
+        buildElementOption,
+      )[0] !== undefined
+        ? editorStore.graphManagerState.usableRuntimes
+            .filter((runtime) =>
```

Comment:
Can we please use `getMappingCompatibleRuntimes` function here to get list of compatible runtimes